cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -128,13 +128,19 @@ func (c *client) EnsureNamespace(namespaceName string) error {
 		return errors.Wrap(err, "error creating core clientset")
 	}
 
+	namespaces, err := clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
```

Comment:
Please use `Get` instead of `List` to check for a single namespace. You can use `apierrors.IsNotFound(err)` to distinguish between a 404 not found and other errors.